### PR TITLE
Added all maven dependencies offline

### PR DESCRIPTION
**Pull request by palla714**
#### What is this PR About?
This will add all maven dependencies offline

#### How do we test this?
Here in this Jenkinsfile we can add the agent to test 

https://github.com/WHOAcademy/WHOA-LXP-TestAutomationSuite/blob/feature-maven/Jenkinsfile

cc: @redhat-cop/day-in-the-life

### CODE REVIEW

File: jenkins-agents/jenkins-agent-mvn/Dockerfile [hunk 1]
```diff
@@ -1,2 +1,29 @@
+#FROM quay.io/openshift/origin-jenkins-agent-maven:4.6
+#ADD settings.xml $HOME/.m2/settings.xml
+#WORKDIR $HOME
+#COPY pom.xml ./
+#RUN mvn -e org.apache.maven.plugins:maven-dependency-plugin:2.10:tree -Dverbose=true
+#RUN mvn -e dependency:go-offline -B
+#RUN mvn package 
+#USER 1001
+
```
Comment (krishnamadhavan) [inline jenkins-agents/jenkins-agent-mvn/Dockerfile:9]: Remove these comment lines if they are not necessary.

File: jenkins-agents/jenkins-agent-mvn/Dockerfile [hunk 2]
```diff
@@ -1,2 +1,29 @@
...
 FROM quay.io/openshift/origin-jenkins-agent-maven:4.6
-ADD settings.xml $HOME/.m2/settings.xml
+ADD settings.xml $HOME/settings.xml
```
Comment (krishnamadhavan) [inline jenkins-agents/jenkins-agent-mvn/Dockerfile:11]: Why did we change this from `$HOME/.m2/settings.xml` to `$HOME/settings.xml` ?

File: jenkins-agents/jenkins-agent-mvn/Dockerfile [hunk 3]
```diff
@@ -1,2 +1,29 @@
...
+
+ENV HOME=/home/
```
Comment (krishnamadhavan) [inline jenkins-agents/jenkins-agent-mvn/Dockerfile:13]: In the above command we already have a `HOME` variable. Why are we overriding the value again ?

File: jenkins-agents/jenkins-agent-mvn/Dockerfile (hunk 2 above)
Comment (palla714) [reply jenkins-agents/jenkins-agent-mvn/Dockerfile:11]: image
  
With the .m2 folder getting this error , so removed it.

File: jenkins-agents/jenkins-agent-mvn/Dockerfile (hunk 3 above)
Comment (palla714) [reply jenkins-agents/jenkins-agent-mvn/Dockerfile]: I have removed overridden codes.

File: jenkins-agents/jenkins-agent-mvn/Dockerfile (hunk 1 above)
Comment (palla714) [reply jenkins-agents/jenkins-agent-mvn/Dockerfile]: Comment lines are removed.

File: jenkins-agents/jenkins-agent-mvn/Dockerfile (hunk 2 above)
Comment (krishnamadhavan) [reply jenkins-agents/jenkins-agent-mvn/Dockerfile:2]: I don't think we can remove it. Can you check how to fix the error without removing it ?

Comment (palla714) [reply jenkins-agents/jenkins-agent-mvn/Dockerfile:4]: I have reset the HOME variable path as the path set in the image is corrupted and does not provide access to download all plugins required. It threw error even after adding the plugin in pom.xml which is as per maven document. 

After resetting the HOME variable for the image, the error resolved. We could use the .m2 folder for storing all dependencies .

Comment (palla714) [reply jenkins-agents/jenkins-agent-mvn/Dockerfile:4]: image

**Review by krishnamadhavan — APPROVED** [review]

**Pull request merged**